### ISSUE TITLE
Peer discovery issue

### ISSUE THREAD

**Issue by phahulin** [paritytech/parity#6047]
Hello,

I try to start a test network with authority round on 3 azure vms: let's call them bootnode=mining node `b` and 2 clients `c1`, `c2` that only sync with the network and don't mine.

First I start parity on `b`, copy its enode url to node.toml of the `c1`, start parity, copy enode to node.toml on `c2` and start parity there. I end up with `b` having two peers, but both `c1` and `c2` each has only a single peer `b`.
Now I simply restart `c1` or `c2`, i.e. hit <kbd>ctrl+c</kbd> and rerun the last command. All three nodes now properly connect with each other: `b` has 2 peers, `c1` has 2 peers and `c2` has 2 peers.

I also noticed that
1. if you start `c1` and then very quickly start `c2`, so that both are started almost simultaneously, they connect properly the first time.
2. if `sudo parity...` is used to start any node or all of them, they connect properly the first time.

Please advise, if there might be some problem with user rights or configuration.

Here is spec.json (based on kovan's genesis.json)
```
{
  "name": "OraclesPoA",
  "engine": {
    "authorityRound": {
      "params": {
        "gasLimitBoundDivisor": "0x400",
        "stepDuration": "5",
        "validators": {
          "list": ["0xDd0BB0e2a1594240fED0c2f2c17C1E9AB4F87126"]
        }
      }
    }
  },
  "params": {
    "maximumExtraDataSize": "0x20",
    "minGasLimit": "0x1388",
    "networkID": "0xC0FFEE"
  },
  "genesis": {
    "seal": {
      "authorityRound": {
        "step": "0x0",
        "signature": "0x0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
      }
    },
    "difficulty": "0x20000",
    "gasLimit": "0x989680"
  },
  "accounts": {
    "0x0000000000000000000000000000000000000001": {
      "balance": "1",
      "builtin": {
        "name": "ecrecover",
        "pricing": {
          "linear": {
            "base": 3000,
            "word": 0
          }
        }
      }
    },
    "0x0000000000000000000000000000000000000002": {
      "balance": "1",
      "builtin": {
        "name": "sha256",
        "pricing": {
          "linear": {
            "base": 60,
            "word": 12
          }
        }
      }
    },
    "0x0000000000000000000000000000000000000003": {
      "balance": "1",
      "builtin": {
        "name": "ripemd160",
        "pricing": {
          "linear": {
            "base": 600,
            "word": 120
          }
        }
      }
    },
    "0x0000000000000000000000000000000000000004": {
      "balance": "1",
      "builtin": {
        "name": "identity",
        "pricing": {
          "linear": {
            "base": 15,
            "word": 3
          }
        }
      }
    },
    "0xDd0BB0e2a1594240fED0c2f2c17C1E9AB4F87126": {
      "balance": "10000000000000000000000"
    }
  }
}
```
Here is node.toml from `b`:
```
[parity]
chain = "spec.json"
base_path = "parity"
[network]
nat="extip:$PUBLIC_IP"
port = 30300
discovery=true
[rpc]
cors = "all"
interface = "all"
hosts = ["all"]
port = 8540
apis = ["web3", "eth", "net", "personal", "parity", "parity_set", "traces", "rpc", "parity_accounts"]
[ui]
interface = "0.0.0.0"
port = 8180
[dapps]
cors = "all"
interface = "0.0.0.0"
hosts = ["all"]
port = 8080
[account]
password = ["node.pwd"]
[mining]
force_sealing = true
engine_signer = "0xDd0BB0e2a1594240fED0c2f2c17C1E9AB4F87126"
reseal_on_txs = "none"
[footprint]
tracing = "on"
pruning = "archive"
fat_db = "on"
```
`b` is started via `$ parity -lengine --config "node.toml" --ui-no-validation`.
When this node is started without `sudo`, the following warning appears in log:
```
main WARN diskmap  Couldn't open disk map for writing: Permission denied (os error 13)
```
but the process continues.

Here is node.toml from `c1` (or `c2`, they differ only in $PUBLIC_IP):
```
[parity]
chain = "spec.json"
base_path = "parity"
[network]
nat="extip:$PUBLIC_IP"
bootnodes=["enode://...@$B_NODE_PUBLIC_IP:30300"]
port = 30300
discovery=true
[rpc]
cors = "all"
interface = "all"
hosts = ["all"]
port = 8540
apis = ["web3", "eth", "net", "personal", "parity", "parity_set", "traces", "rpc", "parity_accounts"]
```
these nodes are started via `$ parity --config "node.toml"`

**Comment by arkpar** [comment]
Could you reproduce this with `-l discovery=trace,network=trace` on all three nodes and share logs?

**Comment by phahulin** [comment]
Here you go
[bootnode.txt](https://github.com/paritytech/parity/files/1144927/bootnode.txt)
[client1.txt](https://github.com/paritytech/parity/files/1144928/client1.txt)
[client2.txt](https://github.com/paritytech/parity/files/1144929/client2.txt)

What I did:
1. bootnode (...191.13) started around 10:56
2. client1 (...36.180) started around 10:58
3. client2 (...32.252) started around 10:59
then I waited about 5 minutes. you can see client1 and client2 have a single peer.
4. client2 restarted around 11:05 to no effect
5. client2 restarted again around 11:06 (usually they connect instantly, so I thought a minute is long enough) to no effect
6. then I decided to restart client1 instead, around 11:08, this time they reconnected properly, you can see each node has 2 peers now
7. I waited a bit longer and shut down bootnode at 11:10 to check if client nodes would stay connected, which they did.
8. then I shutdown client1 around 11:11
9. then I shut down client2 as well

**Comment by phahulin** [comment]
Maybe there's something else I can check?

**Comment by arkpar** [comment]
Identified as a bug. Expect to be fixed soon.

**Comment by phahulin** [comment]
Seems to be working OK.
Can this also be included in the next release (1.6.10, I guess)?

**Comment by arkpar** [comment]
will make it into 1.7.0 soon
